lib/auth/auth_with_roles_test.go

```diff
@@ -3943,114 +3943,92 @@ func TestGenerateHostCert(t *testing.T) {
 	}
 }
 
-// TestLocalServiceRolesHavePermissionsForUploaderService verifies that all of Teleport's
-// builtin roles have permissions to execute the calls required by the uploader service.
-// This is because only one uploader service runs per Teleport process, and it will use
-// the first available identity.
-func TestLocalServiceRolesHavePermissionsForUploaderService(t *testing.T) {
+// TestInstanceRoleCanUpload verifies that Teleport's Instance role
+// has permissions to execute the calls required by the uploader service.
+func TestInstanceRoleCanUpload(t *testing.T) {
```

Comment:
Testing this is good, but effectively this test is verifying that the node role has the permission to upload.  Instance isn't a role in the sense of `RoleAuth` or `RoleNode`, its a placeholder used to indicate that a cert contains multiple system roles.  An instance cert has the combined permissions of all true system roles that are listed on the cert, so wether or not an instance cert will be able to upload depends on wether or not its constituent system roles are able to upload.

Consider making this a sub-test in a for-loop that re-runs it for an instance identity with each system role that we expect should be allowed to upload.